packages/gatsby/src/commands/develop-process.ts

```diff
@@ -210,57 +230,104 @@ module.exports = async (program: IProgram): Promise<void> => {
       },
       doingEverythingElse: {
         invoke: {
-          src: async ({ workerPool, store, app }): Promise<void> => {
+          src: async (): Promise<void> => {
             // All the stuff that's not in the state machine yet
 
-            await writeOutRequires({ store })
             boundActionCreators.setProgramStatus(
               ProgramStatus.BOOTSTRAP_QUERY_RUNNING_FINISHED
             )
 
             await db.saveState()
 
-            await waitUntilAllJobsComplete()
-            requiresWriter.startListener()
             db.startAutosave()
-            queryUtil.startListeningToDevelopQueue({
-              graphqlTracing: program.graphqlTracing,
-            })
-            queryWatcher.startWatchDeletePage()
-
-            await startWebpackServer({ program, app, workerPool, store })
+          },
+          onDone: [
+            {
+              target: `startingDevServers`,
+              cond: ({ compiler }: IBuildContext): boolean => !compiler,
```

Comment:
Why are we checking for compiler here? Shouldn't it be part of a first run instead of doing it every time with a condition?

Reply:
It's effectively the same, but yes checking for `firstRun` would be more explicit (or using a different machine for first run as suggested)